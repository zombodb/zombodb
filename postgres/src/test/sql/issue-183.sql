--
-- should return just id=2,3,4
--
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1';
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[1]';
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[[1]]';

--
-- should return all 4 rows
--
SELECT id FROM products WHERE id <> 1 OR id <> 2 OR id <> 3 ORDER BY id;
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1 OR id<>2 OR id <> 3' ORDER BY id;

--
-- should return just id=4
--
SELECT id FROM products WHERE id <> 1 AND id <> 2 AND id <> 3;
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1 AND id<>2 AND id <>3';
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[1,2,3]';
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[[1,2,3]]';

