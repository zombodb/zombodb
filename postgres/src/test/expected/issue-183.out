--
-- should return just id=2,3,4
--
SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1';
 id 
----
  2
  3
  4
(3 rows)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[1]';
 id 
----
  2
  3
  4
(3 rows)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[[1]]';
 id 
----
  2
  3
  4
(3 rows)

--
-- should return all 4 rows
--
SELECT id FROM products WHERE id <> 1 OR id <> 2 OR id <> 3 ORDER BY id;
 id 
----
  1
  2
  3
  4
(4 rows)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1 OR id<>2 OR id <> 3' ORDER BY id;
 id 
----
  1
  2
  3
  4
(4 rows)

--
-- should return just id=4
--
SELECT id FROM products WHERE id <> 1 AND id <> 2 AND id <> 3;
 id 
----
  4
(1 row)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>1 AND id<>2 AND id <>3';
 id 
----
  4
(1 row)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[1,2,3]';
 id 
----
  4
(1 row)

SELECT id FROM products WHERE zdb('products', ctid) ==> 'id<>[[1,2,3]]';
 id 
----
  4
(1 row)

